In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
from wrangle import get_db_url

In [2]:
sql_query = """
        SELECT *
        from properties_2017
        join propertylandusetype using (propertylandusetypeid)
        
        where	propertylandusetypeid = '261';
                """

In [2]:
sql_query2 = """
            SELECT yearbuilt, calculatedfinishedsquarefeet, lotsizesquarefeet, fips, regionidzip, bedroomcnt, bathroomcnt,hashottuborspa, garagecarcnt, fireplacecnt,poolcnt
from properties_2017
JOIN propertylandusetype USING (propertylandusetypeid)
JOIN predictions_2017 ON properties_2017.id = predictions_2017.id
WHERE propertylandusetype.propertylandusedesc = "Single Family Residential" 
AND predictions_2017.transactiondate LIKE "2017%%";
                    """

In [14]:
sql_query3 = """
            select bedroomcnt,
    bathroomcnt,
    calculatedfinishedsquarefeet,
    taxvaluedollarcnt,
    yearbuilt,
    fips,
    buildingqualitytypeid,
    decktypeid,
    fireplacecnt,
    garagecarcnt,
    poolcnt,
    regionidzip,
    propertylandusedesc,
    airconditioningdesc,
    buildingclassdesc,
    architecturalstyledesc,
    transactiondate
FROM properties_2017 
LEFT JOIN propertylandusetype  USING (propertylandusetypeid)
LEFT JOIN airconditioningtype  USING (airconditioningtypeid)
LEFT JOIN buildingclasstype  USING (buildingclasstypeid)
LEFT JOIN architecturalstyletype  USING (architecturalstyletypeid)
RIGHT JOIN predictions_2017  USING (parcelid)
WHERE propertylandusedesc = 'Single Family Residential'
AND predictions_2017.transactiondate LIKE "2017%%";
"""

In [72]:
sql_query4 = """
            SELECT yearbuilt, taxvaluedollarcnt, calculatedfinishedsquarefeet, lotsizesquarefeet, fips, regionidzip, bedroomcnt, bathroomcnt,hashottuborspa, garagecarcnt, fireplacecnt,poolcnt
FROM properties_2017 
LEFT JOIN propertylandusetype  USING (propertylandusetypeid)
LEFT JOIN airconditioningtype  USING (airconditioningtypeid)
LEFT JOIN buildingclasstype  USING (buildingclasstypeid)
LEFT JOIN architecturalstyletype  USING (architecturalstyletypeid)
RIGHT JOIN predictions_2017  USING (parcelid)
WHERE propertylandusedesc = 'Single Family Residential'
AND predictions_2017.transactiondate LIKE "2017%%";
"""

In [59]:
df = pd.read_sql(sql_query4, get_db_url("zillow"))
#df = df.dropna()

In [60]:
pd.set_option('display.max_columns', None)
df.head()

yearbuilt  taxvaluedollarcnt  calculatedfinishedsquarefeet  \
0     1998.0          1023282.0                        3100.0   
1     1967.0           464000.0                        1465.0   
2     1962.0           564778.0                        1243.0   
3     1970.0           145143.0                        2376.0   
4     1950.0           773303.0                        2962.0   

   lotsizesquarefeet    fips  regionidzip  bedroomcnt  bathroomcnt  \
0             4506.0  6059.0      96978.0         4.0          3.5   
1            12647.0  6111.0      97099.0         2.0          1.0   
2             8432.0  6059.0      97078.0         3.0          2.0   
3            13038.0  6037.0      96330.0         4.0          3.0   
4            63000.0  6037.0      96293.0         4.0          3.0   

   hashottuborspa  garagecarcnt  fireplacecnt  poolcnt  
0             NaN           2.0           NaN      NaN  
1             NaN           1.0           1.0      NaN  
2             NaN           2.0           NaN      1.0  
3             NaN           NaN           NaN      1.0  
4             NaN           NaN           NaN      1.0

In [47]:
df.info(show_counts=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 52441 entries, 0 to 52440
Data columns (total 12 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   yearbuilt                     52325 non-null  float64
 1   taxvaluedollarcnt             52440 non-null  float64
 2   calculatedfinishedsquarefeet  52359 non-null  float64
 3   lotsizesquarefeet             52072 non-null  float64
 4   fips                          52441 non-null  float64
 5   regionidzip                   52415 non-null  float64
 6   bedroomcnt                    52441 non-null  float64
 7   bathroomcnt                   52441 non-null  float64
 8   hashottuborspa                52441 non-null  float64
 9   garagecarcnt                  52441 non-null  float64
 10  fireplacecnt                  52441 non-null  float64
 11  poolcnt                       52441 non-null  float64
dtypes: float64(12)
memory usage: 4.8 MB


In [71]:
df.bedroomcnt.value_counts()

3.0     23265
4.0     15111
2.0      8265
5.0      3915
6.0       627
1.0       578
7.0       103
0.0        54
8.0        24
9.0         8
10.0        2
11.0        1
Name: bedroomcnt, dtype: int64

In [70]:
df.shape

(51953, 12)

**TAKEAWAYS**
- Raw dataset has 52441 rows and  13 columns
- Need to `poolcnt` fill null values with 0 
- Need to fill `garagecarcnt` null values with 0
- Need to fill `fireplacecnt` null values with 0
- Need to fill `hashottuborspa` null values with 0

- **Will determine what  values to  convert to integer**
- **Will determine what Null values to drop afterwards**
- **Encode or get dummies for categorial data**

In [61]:
df["poolcnt"].fillna(0, inplace = True)

In [62]:
df["garagecarcnt"].fillna(0, inplace = True)

In [63]:
df["fireplacecnt"].fillna(0, inplace = True)

In [64]:
df["hashottuborspa"].fillna(0, inplace = True)

In [46]:
df[df.isna().any(axis=1)]

yearbuilt  taxvaluedollarcnt  calculatedfinishedsquarefeet  \
50        1885.0           232000.0                        1046.0   
79        2006.0           145452.0                        1908.0   
124          NaN           830145.0                           NaN   
372       2014.0           454832.0                        1689.0   
451       1963.0           501488.0                        1195.0   
...          ...                ...                           ...   
51656     1979.0           176546.0                         764.0   
51708     2008.0          1588348.0                        5197.0   
51734     1937.0           235612.0                        1032.0   
52224     2012.0          2912659.0                        5945.0   
52429        NaN            34124.0                         892.0   

       lotsizesquarefeet    fips  regionidzip  bedroomcnt  bathroomcnt  \
50                   NaN  6037.0      96506.0         2.0          1.0   
79                   NaN  6037.0      97318.0         4.0          3.0   
124              39184.0  6059.0      97078.0         0.0          0.0   
372                  NaN  6037.0      96173.0         3.0          2.0   
451                  NaN  6059.0      96969.0         2.0          2.0   
...                  ...     ...          ...         ...          ...   
51656                NaN  6059.0      96985.0         1.0          1.0   
51708                NaN  6037.0      96522.0         5.0          6.0   
51734                NaN  6037.0      96109.0         2.0          1.0   
52224                NaN  6037.0      96449.0         5.0          6.0   
52429             5201.0  6037.0      96020.0         0.0          0.0   

       hashottuborspa  garagecarcnt  fireplacecnt  poolcnt  
50                0.0           0.0           0.0      0.0  
79                0.0           0.0           0.0      0.0  
124               0.0           0.0           0.0      0.0  
372               0.0           0.0           0.0      0.0  
451               0.0           1.0           1.0      0.0  
...               ...           ...           ...      ...  
51656             0.0           1.0           0.0      1.0  
51708             0.0           0.0           0.0      0.0  
51734             0.0           0.0           0.0      0.0  
52224             0.0           0.0           0.0      1.0  
52429             0.0           0.0           0.0      0.0  

[488 rows x 12 columns]

**filter  dataframe to have 

In [65]:
df.dropna(inplace=True)


In [68]:
df.shape

(51953, 12)